lib/base-compiler.js

```diff
@@ -291,23 +297,31 @@ Compile.prototype.compile = function (source, options, backendOptions, filters)
                 }
                 return result;
             })
-            .then(_.bind(self.postProcessAsm, self))
+            .then(_.bind(function(result){return self.postProcessAsm(result, filters);}, self))
             .then(function (result) {
                 if (result.okToCache) self.env.cachePut(key, result);
                 return result;
             });
     });
 };
 
-Compile.prototype.postProcessAsm = function (result) {
+Compile.prototype.postProcessAsm = function (result, filters) {
+    var self = this;
```

Comment:
We've tried our best to avoid using the `self` idiom. If possible, revert to using `this` and add a `bind`. In the code below, using `.then(_.bind(function (demangleResult {...}, this);` should do the trick. Javascript is nasty like this, and we'd definitely enjoy the benefits of typescript here. (CC @jaredwy who I think was considering starting the move to Typescript). Or else we should start moving to arrow functions. Anyway, for now, if you can easily phrase it using `this` and not `self`, that would be better.

Reply:
changed to bind for base-compiler only.